app/src/main/java/ru/prometeydev/movie/model/MoviesPagingSource.kt

```diff
@@ -0,0 +1,31 @@
+package ru.prometeydev.movie.model
+
+import androidx.paging.PagingSource
+import retrofit2.HttpException
+import ru.prometeydev.movie.model.local.Movie
+import java.io.IOException
+import java.lang.Exception
+
+class MoviesPagingSource(
+    private val repository: MoviesRepository
+) : PagingSource<Int, Movie>() {
+
+    override suspend fun load(params: LoadParams<Int>): LoadResult<Int, Movie> {
+        val page = params.key ?: STARTING_PAGE_INDEX
+        return try {
+            val movies = repository.loadMovies(page)
+            LoadResult.Page(
+                data = movies,
+                prevKey = if (page == STARTING_PAGE_INDEX) null else page - 1,
```

Comment:
Is null marker about no data? Is it library expectations or your implementation?

Reply:
This implementation example was borrowed by me from codelab https://developer.android.com/codelabs/android-paging#4
those. if we are on the first page, then the previous key must be null.

Reply:
The same principle is applied in this article https://proandroiddev.com/how-to-use-the-paging-3-library-in-android-5d128bb5b1d8